lib/src/components/NxTabs/__tests__/NxTabs.test.tsx

```diff
@@ -5,103 +5,76 @@
 import NxTabList from '../NxTabList';
 import NxTab from '../NxTab';
 import NxTabPanel from '../NxTabPanel';
 
-describe('NxTabs', function () {
-  it('renders with a unique id each time', function () {
-    const firstTabs = shallow(
-      <NxTabs onTabSelect={() => {}} activeTab={0}>
-        <NxTabList>
-          <NxTab>Tabs 0 Tab 0</NxTab>
-        </NxTabList>
-      </NxTabs>
-    );
-    const secondTabs = shallow(
-      <NxTabs onTabSelect={() => {}} activeTab={0}>
-        <NxTabList>
-          <NxTab>Tabs 1 Tab 0</NxTab>
-        </NxTabList>
-      </NxTabs>
-    );
+describe('NxTabs', function() {
+  const minimumProps = {
+    onTabSelect: () => {},
+    children: (
+      <NxTabList>
+        <NxTab>Tabs 0</NxTab>
+      </NxTabList>
+    )
+  };
 
-    expect(firstTabs).not.toHaveProp('id', secondTabs.prop('id'));
-  });
+  const quickRender = rtlRender<NxTabsProps>(NxTabs, minimumProps);
+  const renderEl = rtlRenderElement<NxTabsProps>(NxTabs, minimumProps);
 
-  it('renders with the correct classnames', function () {
-    const component = shallow(
-      <NxTabs onTabSelect={() => {}} activeTab={-1} className="test">
-        <NxTabList>
-          <NxTab>Tab 0</NxTab>
-        </NxTabList>
-        <NxTabPanel>Content 0</NxTabPanel>
-      </NxTabs>
-    );
+  it('merges any passed in className', function() {
+    const tabsWithAddedClassName = renderEl({ activeTab: -1, className: 'foo' });
+    const tabs = renderEl({ activeTab: -1 })!;
+
+    expect(tabsWithAddedClassName).toHaveClass('foo');
 
-    expect(component).toHaveClassName('nx-tabs test');
+    for (const cls of Array.from(tabs.classList)) {
+      expect(tabsWithAddedClassName).toHaveClass(cls);
+    }
   });
 
-  it('renders a generated id', function () {
-    const component = shallow(
-      <NxTabs onTabSelect={() => {}} activeTab={-1}>
-        <NxTabList>
-          <NxTab>Tab 0</NxTab>
-        </NxTabList>
-        <NxTabPanel>Content 0</NxTabPanel>
-      </NxTabs>
-    );
+  it('renders with a unique id each time', function() {
+    const firstTabs = renderEl({ activeTab: 0 })!;
+    const secondTabs = renderEl({ activeTab: 0 })!;
 
-    expect(component.prop('id')).toMatch(/^nx-tabs-\d+$/);
+    expect(firstTabs.id).not.toEqual(secondTabs.id);
   });
 
-  it('renders using as specific id', function () {
-    const component = shallow(
-      <NxTabs onTabSelect={() => {}} activeTab={-1} id="my-tabs">
-        <NxTabList>
-          <NxTab>Tab 0</NxTab>
-        </NxTabList>
-        <NxTabPanel>Content 0</NxTabPanel>
-      </NxTabs>
-    );
+  it('renders a generated id', function() {
+    const component = renderEl({ activeTab: -1 })!;
+    expect(component.id).toMatch(/^nx-tabs-\d+$/);
+  });
 
-    expect(component.prop('id')).toMatch('my-tabs');
+  it('renders using as specific id', function() {
+    const commponent = renderEl({
+      id: 'my-tabs',
+      activeTab: -1
+    })!;
+    expect(commponent.id).toMatch('my-tabs');
   });
```

Comment:
Also test how an explicit id affects the generated ids of the children

Reply:
> Also test how an explicit id affects the generated ids of the children

Updated here: 109736a065caf6fe1852838fb3261f4c5eb097bf